### PR TITLE
Bump compat LossFunctions = "0.9" and address breakage

### LINKED ISSUE #896: Update to LossFunctions.jl v0.9
@ablaom we refactored LossFunctions.jl and changed the order of output and prediction arguments to be more consistent with other ecosystems as you requested at some point in the past. Can you please update this package to use the latest release?

We also removed the ObsDim submodule and the OrdinalMarginLoss as they were not very useful nor very used in downstream projects we maintain.

### PR DESCRIPTION
**Pull request by ablaom**
Closes: #896 #570

### CODE REVIEW

File: src/measures/loss_functions_interface.jl [hunk 1]
```diff
@@ -44,7 +44,7 @@ err_wrap(n) = ArgumentError("Bad @wrap syntax: $n. ")
 
 # We define amacro to wrap a concrete `LossFunctions.SupervisedLoss`
 # type and define its constructor, and to define property access in
-# case of paramters; the macro also defined calling behaviour:
+# case of paramters; the macro also defines calling behaviour:
```
Comment (ToucheSir) [inline src/measures/loss_functions_interface.jl:47]:
```suggestion
# case of parameters; the macro also defines calling behaviour:
```
Drive-by comment since I've been following https://github.com/JuliaAI/MLJBase.jl/issues/570 and am interested in how to effectively use the new LossFunctions interface :)

**Comment by ablaom** [comment]
Thanks @ToucheSir . FYI I am working on a major refactor of the MLJMeasures code, which will live in a separate package, which will better leverage MLUtils.jl.  I'm considering dropping LossFunctions.jl altogether there (see [this issue](https://github.com/JuliaML/LossFunctions.jl/issues/157)) and to just re-implement those functions. Given the infrastructure I have developed, this would be pretty simple, and I'll get multi-target versions of the loss functions for free (working on matrices, tables or arrays.)  I'll also make the basic type-free scalar functions public.
